amundsen_application/static/js/ducks/search/reducer.ts

```diff
@@ -27,7 +30,7 @@ export interface SearchReducerState {
 };
 
 /* ACTIONS */
-export function searchAll(term: string, resource: ResourceType, pageIndex: number): SearchAllRequest {
+export function searchAll(term: string, resource?: ResourceType, pageIndex?: number): SearchAllRequest {
```

Comment:
Why optional as opposed to setting defaults?

Reply:
I went back and forth between using optionals, defaults, and even including auto-select tab in this PR. Ultimately I decided to save auto-select for a followup but there are some leftover changes like this. These don't even need to be optional (yet).